src/DepositManager.mo

```diff
@@ -221,10 +197,33 @@ module {
     /// n - desired number of potential consolidations.
     public func trigger(n : Nat) : async* () {
       for (i in Iter.range(1, n)) {
-        let ?(p, deposit, release) = depositMap.nextLock() else return;
-        underwayFunds += deposit;
-        let result = await* consolidate(p, release);
-        underwayFunds -= deposit;
+        let stack = Data.Queue<Principal>();
+
+        func search() : ?Data.Entry<Principal> {
+          loop {
+            let ?entry = queue.popFront() else return null;
+
+            if (entry.deposit() > feeManager.fee()) {
+              if (entry.locked()) stack.pushBack(entry) else return ?entry;
+            } else {
+              entry.setDeposit(0);
+            };
+          };
+        };
+
+        func addBack() {
+          loop {
+            let ?entry = stack.popBack() else return;
+            queue.pushFront(entry);
+          };
+        };
+
+        let ?entry = search() else return;
+        addBack();
+
+        underwayFunds += entry.deposit();
+        let result = await* consolidate(entry);
+        underwayFunds -= entry.deposit();
```

Comment:
Shall we put the two line that modify `underwayFunds` inside the `consolidate` function next to where we call `entry.setDeposit()`? Then it is clearer how the accounting works and why it is correct.